### PR TITLE
recipe for deepdiff

### PR DESCRIPTION
**Pull request by loriab**
Adding a recipe for https://github.com/seperman/deepdiff . Will contact original developer for permission to include as maintainer.

### CODE REVIEW

File: recipes/deepdiff/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{% set name = "deepdiff" %}
+{% set version = "3.3.0" %}
+{% set md5 = "d16022fbf7b6a1ad401c629b81088614" %}
```
Comment (synapticarbors) [inline recipes/deepdiff/meta.yaml]: Please use the `sha256`: `ecad8e16a96ffd27e8f40c9801a6ab16ec6a7e7e6e6859a7710ba4695f22702c`

File: recipes/deepdiff/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  md5: {{ md5 }}
```
Comment (synapticarbors) [inline recipes/deepdiff/meta.yaml]: use `sha256`.

File: recipes/deepdiff/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+build:
+  noarch: python
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - idna 2.5     # [win]
```
Comment (synapticarbors) [inline recipes/deepdiff/meta.yaml]: I don't think `idna` or  `jsonpickle` are required to actually  build the package.

File: recipes/deepdiff/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    - jsonpickle
+  run:
+    - python
+    - idna 2.5     # [win]
```
Comment (synapticarbors) [inline recipes/deepdiff/meta.yaml]: I couldn't find any reference to `idna` when grepping for it in the source. Is it needed?

File: recipes/deepdiff/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  md5: {{ md5 }}
+
+build:
+  noarch: python
```
Comment (synapticarbors) [inline recipes/deepdiff/meta.yaml]: If `idna` is in fact a dependency (see other comment) that is platform specific, then `noarch` isn't appropriate:

http://conda-forge.readthedocs.io/en/latest/meta.html#building-noarch-packages

**Comment by loriab** [comment]
Thanks, @synapticarbors, for review and the heads-up on the true cause of the win/py35/idna problem.

Wrt your other comments, md5 converted, build dep jsonpickle removed, and indeed idna dep was just my grasp to fix the appveyor fails.

Fyi, maintainer permission is sought [here](https://github.com/seperman/deepdiff/issues/89).

**Comment by jakirkham** [comment]
Toggling for CIs now that the `idna` issue should be fixed.

**Pull request merged**